src/components/article/SingleArticle.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { BiSolidLike } from "react-icons/bi";
+import { getArticleById } from "../../utils/api";
+import { NewsDataContext } from "../../data/NewData";
+
+const SingleArticle = () => {
+    const [singleArticle, setSingleArticle] = useState(null);
+    const [userAvatar, setUserAvatar] = useState("");
+    const [isMore, setIsMore] = useState(false);
+    const { userList } = useContext(NewsDataContext);
+    const { article_id } = useParams();
+
+    useEffect(() => {
+        getArticleById(article_id).then((article) => {
+            setSingleArticle(article);
+        });
+    }, []);
```

Comment:
While this request is being made is the user given any indication that something is happening?

Reply:
Ok, I see I will add that. Many Thanks.